### PR TITLE
feat: set read-ahead for nfs mount

### PR DESCRIPTION
**Pull request by cvvz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
#### io wait is high and cpu usage is low
image
#### cache hits times is low
image

#### vfs_read execution time
image

#### io wait is 0and cpu usage is higher than before
image

#### cache hits times is higher
image

#### vfs_read execution time
image

**Release note**:
```
none
```

### CODE REVIEW

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Comment by cvvz** [comment]
@andyzhangx PTAL

**Comment by andyzhangx** [comment]
> @andyzhangx PTAL

@cvvz it works now? what's the change later?

**Comment by cvvz** [comment]
Yes, absolute path of `awk` is different in different Linux distribution and version, AKS is using Ubuntu 18.04 and `/bin/awk` is not right executable path.

**Comment by cvvz** [comment]
This script is now work for different Linux distribution version

**Comment by cvvz** [comment]
They have updated the documentation to reflect this.
[Linux NFS read-ahead best practices for Azure NetApp Files - Session slots and slot table entries | Microsoft Learn](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Flearn.microsoft.com%2Fen-us%2Fazure%2Fazure-netapp-files%2Fperformance-linux-nfs-read-ahead%23how-to-persistently-set-read-ahead-for-nfs-mounts&data=05%7C01%7Cweizhichen%40microsoft.com%7C05fccaeb2afe4f3261b008daa2d1ba85%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638001318944233443%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=BUCO84Adxnla4LqAOETBHotFBXOa0a9%2BTPgTsTgoCTE%3D&reserved=0)

File: pkg/blobfuse-proxy/init.sh [hunk 1]
```diff
@@ -89,4 +90,10 @@ then
   sed -i 's/PRUNEFS="NFS/PRUNEFS="fuse blobfuse NFS/g' ${updateDBConfigPath}
   echo "after change:"
   cat ${updateDBConfigPath}
-fi
\ No newline at end of file
+fi
+
+AWK_PATH=$(which awk)
```
Comment (andyzhangx) [inline pkg/blobfuse-proxy/init.sh:95]: add a new flag to control whether set read-ahead for nfs mount, by default is true

File: pkg/blobfuse-proxy/init.sh [hunk 2]
```diff
@@ -89,4 +91,13 @@ then
   sed -i 's/PRUNEFS="NFS/PRUNEFS="fuse blobfuse NFS/g' ${updateDBConfigPath}
   echo "after change:"
   cat ${updateDBConfigPath}
+fi
+
+if [ "${SET_READ_AHEAD_SIZE}" = "true" ]
+then
+  AWK_PATH=$(which awk)
```
Comment (andyzhangx) [inline pkg/blobfuse-proxy/init.sh:98]: add an echo in the log to show that read-ahead is set for nfs mount

**Review by andyzhangx — APPROVED** [review]
/approve
pls squash all commits

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**